demo/plugins/lookup_plugins/conjur_variable.py

```diff
@@ -0,0 +1,156 @@
+    # before returning if necessary.
+    def get_header_value(self):
+        if time() >= self.refresh_time:
+            self.refresh()
+
+        return 'Token token="%s"' % (self.token)
+
+
+def load_conf(conf_path):
+    conf_path = os.path.expanduser(conf_path)
+
+    if not os.path.isfile(conf_path):
+        return {}
+
+    with open(conf_path, 'r') as conf_file:
+        try:
+            return yaml.load(conf_file)
+        except yaml.YAMLError as e:
+            exit_error(e)
+
+
+def load_identity(identity_path, appliance_url):
+    identity_path = os.path.expanduser(identity_path)
+
+    if not os.path.isfile(identity_path):
+        return {}
+
+    try:
+        identity = netrc(identity_path)
+        id, _, api_key = identity.authenticators('{}/authn'.format(appliance_url))
+        if not id or not api_key:
+            return {}
+
+        return dict(id=id, api_key=api_key)
+    except:
+        pass
+
+    return {}
+
+
+# merge_dict
+# Merges all key values in a variable list of dictionaries
+def merge_dict(*arg):
+    ret = {}
+    for a in arg:
+        trimmed_dict = dict((k, v) for k, v in a.iteritems() if v)
+        ret = dict(ret.items() + trimmed_dict.items())
+    return ret
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+
+        # Load Conjur configuration
+        conf = merge_dict(load_conf('/etc/conjur.conf'),
+                          load_conf('~/.conjurrc'),
+                          dict(
+                              account = environ.get('CONJUR_ACCOUNT'),
+                              appliance_url = environ.get("CONJUR_APPLIANCE_URL"),
+                              cert_file = environ.get('CONJUR_CERT_FILE')
+                          ))
+
+        os.system('echo conf: %s >> /tmp/conjur_variable.txt' % conf)
+
+        if not conf:
+            exit_error('Failed to load a Conjur configuration.\n'
+                       'Make sure a configuration exists at environment variables or at either of the following paths:\n'
+                       '~/.conjurrc'
+                       '/etc/conjur.conf\n')
+
+        # Load Conjur identity
+        identity = merge_dict(
+            load_identity('/etc/conjur.identity', conf['appliance_url']),
+            load_identity('~/.netrc', conf['appliance_url']),
+            dict(
+                api_key = environ.get('CONJUR_AUTHN_API_KEY'),
+                id = environ.get('CONJUR_AUTHN_LOGIN')
+            ))
+
+        os.system('echo identity: %s >> /tmp/conjur_variable.txt' % identity)
+
+        if not identity:
+            exit_error('Failed to load a Conjur identity.\n'
+                       'Make sure an identity exists at environment variables or at either of the following paths:\n'
+                       '~/.netrc' 
+                       '/etc/conjur.identity\n')
+
+        # Load our certificate for validation
+        # ssl_context = ssl.create_default_context()
```

Comment:
I think we should merge with master with the SSL.
Also maybe we should add an option to use HTTP / HTTPS and not only one of them

Reply:
I agree, let's get SSL on master.

HTTP is only good for testing. The default should be HTTPS.